src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -31,18 +34,33 @@ public DefaultSinkRecordConverter(Map<String, String> kafkaConfig) {
 		if (val != null && val.trim().length() > 0) {
 			format = Format.valueOf(val.toUpperCase());
 		}
-
 		val = kafkaConfig.get(MarkLogicSinkConfig.DOCUMENT_MIMETYPE);
 		if (val != null && val.trim().length() > 0) {
 			mimeType = val;
 		}
 	}
-
+	
 	@Override
 	public DocumentWriteOperation convert(SinkRecord sinkRecord) {
-		return documentWriteOperationBuilder.build(toContent(sinkRecord));
+		return documentWriteOperationBuilder.build(toContent(sinkRecord), addTopicToCollections(sinkRecord.topic(), addTopicToCollections ) );
 	}
 
+	/**
+	 * 
+	 *
+	 * @param topic, addTopicToCollections
+	 * @return  metadata 
+	 */
+	public DocumentMetadataHandle addTopicToCollections (String topic, Boolean addTopicToCollections) {
+		DocumentMetadataHandle metadata = new DocumentMetadataHandle();
+			if (addTopicToCollections) {
+				metadata.getCollections().addAll(topic);
+			} else {
```

Comment:
No need for this else method, since nothing's being added. Also, just call "add" above instead of "addAll" - "addAll" implies that "topic" is an array, but it's a single string.